### ISSUE TITLE
Challenge 26 - Fire Forecasting

### ISSUE THREAD

**Issue by EsperanzaCuartero** [ECMWFCode4Earth/challenges_2023#13]
# Challenge 26 - Fire Forecasting

> **Stream 2** - Machine Learning for Earth Science 

### Goal
Develop an application using ML for forecasting of vegetation fires in Europe for one hour to five days ahead

### Mentors and skills
* **Mentors:** Johannes Kaiser, Mark Parrington, Miha Razinger, Mihai Alexe and Siham El Garroussi
* **Skills required:**
  * Python programming (numpy, pandas, xarray, matplotlib, cartopy …)
  * Machine learning model development and validation (scikit-learn, Pytorch, Tensorflow and/or Jax)
  * Processing and visualization of scientific data (netCDF, HDF, GRIB, …)

<br>

> <b> *Note: Only nationals from European Union (EU) Member States and countries associated with EU’s Space Programme (currently Iceland and Norway) are eligible to participate (see [Terms and Conditions](https://codeforearth.ecmwf.int/terms-and-conditions)).* </b>

<hr>

### Challenge description
CAMS is providing daily ensemble forecasts of air quality in the European domain. The influence of smoke from open vegetation fires like forest fires, peat fires and agricultural waste burning is modelled based on emission estimates generated by the Global Fire Assimilation System (GFAS). GFAS assimilates satellite-based observations of the infrared radiation emitted by vegetation fires (aka landscape fires) to produce daily and hourly estimates of biomass burning emissions. A key challenge is that large observation gaps exist due to cloud cover and discrete satellite overpass times, while the combustion rate displays strong diurnal and day-to-day variability. Currently, the persistence of detected daytime and night-time fires is assumed, which is a major source of errors in GFAS, the 5-day forecasts of European air quality produced by CAMS and, to a lesser extent, the global CAMS forecasts and reanalyses.

It is well-established that the development of existing large fires primarily depends on wind, precipitation, fuel type, slope, fuel moisture, air humidity and other meteorological and geographical quantities. This is exploited, for example, by various fire danger indices provided by meteorological services and the Copernicus Emergency Management Service (CEMS). These indices are traditionally calculated with daily resolution based on the meteorological conditions at local noon. Despite knowledge of many factors that influence the development of already burning landscape fires, their quantitative forecasting at hourly and daily resolutions has not been achieved yet due to the complexity of the problem and incomplete knowledge of all factors of influence at high spatial resolution.

The aim of this project is to use machine learning to overcome the difficulties in forecasting the development of already detected landscape fires (as represented in the GFAS analyses) with machine learning techniques in order to extend GFAS with a first version 5-day forecasts of fire activity and smoke emissions for the European domain. The target resolutions are 1 hour temporally and 0.1 deg spatially. A secondary objective may be to improve GFAS analyses and forecasts globally by using such fire activity forecasts as model during the Kalman filtering.

The project will be divided into the following stages:
1) **Data collection**: Collection of pre-processed satellite observations and assimilated fields of fire activity from the operational GFAS production and relevant meteorological conditions from the operational weather forecasts of ECMWF in the European domain.
2) **Clustering**: Extraction of training and validation datasets from data collection. Resources from https://www.aireo.net will facilitate the creations of a training dataset that is useful for the both the next stage of the project and applications beyond this project.
3) **Forecast model identification and training**: This is the main stage and will require investigating different ML techniques and configurations to identify the appropriate approach for forecasting fire activity based on the latest GFAS estimate of the current fire activity.
4) **Independent validation**: Validation of the forecasting model to determine its accuracy.
5) **Consolidation**: The result should be delivered as stand-alone software for (a) training of the fire forecasting model that can be applied to any region on Earth and any future update of the satellite observations assimilated in GFAS and (b) actual forecasting of fire activity based on the latest operational GFAS products and weather forecasts. The stand-alone software will subsequently be integrated into the GFAS processing chain.

**Comment by r-maiwald** [comment]
Hi,
I am interested in in this challenge and I am drafting a proposal. I was wondering if it will be possible to use ressources from the ECMWF for the training of the ML models?

**Comment by KaiserSF** [comment]
Hi there,
Yes, you will be able to use computing resources of ECMWF.
Looking forward to your proposal!
Johannes
